src/adapters/net.cpp

```diff
@@ -21,14 +23,65 @@
 POLYBAR_NS
 
 namespace net {
+  typedef enum {
```

Comment:
some of this code is similar to i3status code, unsure if attribution is needed

Reply:
i3status uses the 3-clause BSD license, so if we copy their source code, we need to license that part of the code under the same license, this is not desirable.

So we shouldn't just copy their enum definition. We should also use an enum class here instead of a plain enum.